### PR TITLE
Feature/class weighted dice and ce loss

### PR DESCRIPTION
**Pull request by thijsgelton**
- Added trainer class with a weighted CE loss and a SoftDiceLoss (no weighting)
- Made all configurable through a command line parameter called --trainer_kwargs (see help text for expected format)
- Tested by hand

### CODE REVIEW

File: nnunet/run/run_training.py [hunk 1]
```diff
@@ -23,6 +23,7 @@
 from nnunet.training.network_training.nnUNetTrainerCascadeFullRes import nnUNetTrainerCascadeFullRes
 from nnunet.training.network_training.nnUNetTrainerV2_CascadeFullRes import nnUNetTrainerV2CascadeFullRes
 from nnunet.utilities.task_name_id_conversion import convert_id_to_task_name
+from nnunet.utilities.argparse import ParseJson
```
Comment (nlessmann) [inline nnunet/run/run_training.py:26]: Seems like you are not using this?

Comment (thijsgelton) [reply nnunet/run/run_training.py:26]: Yes, true. I will fix this.

File: nnunet/utilities/shutil_sol.py [hunk 2]
```diff
@@ -33,5 +33,6 @@ def copytree(src, dst, ignore=None):
 
         if len(non_permission_errors) > 0:
             raise shutil.Error(non_permission_errors)
-
+    except FileExistsError:
```
Comment (nlessmann) [inline nnunet/utilities/shutil_sol.py:36]: What's the reason for catching these errors? Shouldn't they be caught by the user if they want to ignore them?

Comment (thijsgelton) [reply nnunet/utilities/shutil_sol.py:36]: This does not  happen when starting a single job, but if you are deploying an interactive session where you alter code, and you pip un/install, then you will receive this message of /home/user/{task} data already being there and the code stops. The only way to not receive the error is to remove the copied files and copy them again. This is 10 minutes unnecessarily wasted.

Comment (nlessmann) [reply nnunet/utilities/shutil_sol.py:36]: Okay - but this should then be accounted for in the place where this function (copytree) is used.

**Review by nlessmann — APPROVED** [review]

**Pull request merged**